stdlib/public/core/Diffing.swift

```diff
@@ -76,34 +79,44 @@ extension RangeReplaceableCollection {
       into target: inout Self,
       contentsOf source: Self,
       from index: inout Self.Index, count: Int
-    ) {
+    ) throws {
       let start = index
-      source.formIndex(&index, offsetBy: count)
+      if !source.formIndex(&index, offsetBy: count, limitedBy: source.endIndex) {
+        throw _ApplicationError.failed
```

Comment:
These throws may throw off people who break on all throws. We can fix this later if it proves annoying.